packages/expo-updates/ios/Tests/EXUpdatesBuildDataTests.m

```diff
@@ -0,0 +1,192 @@
+  
+    NSError *updatesError;
+    [_db addUpdate:update error:&updatesError];
+    if (updatesError) {
+      XCTFail(@"%@", updatesError.localizedDescription);
+      return;
+    }
+
+    NSError *queryError;
+    NSArray<EXUpdatesUpdate *> *allUpdates = [_db allUpdatesWithConfig:_configChannelTest error:&queryError];
+    if (queryError) {
+      XCTFail(@"%@", queryError.localizedDescription);
+      return;
+    }
+    
+    XCTAssertGreaterThan(allUpdates.count, 0);
+  });
+  
+  [EXUpdatesBuildData clearAllUpdatesFromDatabase:_db config:_configChannelTest error:nil];
+  
+  dispatch_sync(_db.databaseQueue, ^{
+    NSError *queryError;
+    NSArray<EXUpdatesUpdate *> *allUpdatesAfter = [_db allUpdatesWithConfig:_configChannelTest error:&queryError];
+    if (queryError) {
+      XCTFail(@"%@", queryError.localizedDescription);
+      return;
+    }
+    
+    XCTAssertEqual(allUpdatesAfter.count, 0);
+  });
+}
+
+- (void)test_ensureBuildDataIsConsistent_buildDataIsNull {
+  // check no updates and build data is set
+
+  dispatch_sync(_db.databaseQueue, ^{
+    NSError *error;
+    NSDictionary *staticBuildData = [EXUpdatesBuildData getBuildDataFromDatabase:_db scopeKey:scopeKey error:&error];
+    XCTAssertNil(staticBuildData);
+
+    NSArray<EXUpdatesUpdate *> *allUpdates = [_db allUpdatesWithConfig:_configChannelTest error:&error];
+    XCTAssertEqual(allUpdates.count, 0);
+    XCTAssertNil(error);
+  });
+
+  NSError *error;
+  [EXUpdatesBuildData ensureBuildDataIsConsistent:_db config:_configChannelTest error:&error];
+  XCTAssertNil(error);
+
+  
+  dispatch_sync(_db.databaseQueue, ^{
+      NSError *error;
+      NSDictionary *newStaticBuildData = [EXUpdatesBuildData getBuildDataFromDatabase:_db scopeKey:scopeKey error:&error];
+      XCTAssertNotNil(newStaticBuildData);
+      XCTAssertNil(error);
+  });
+
+}
+
+- (void)test_ensureBuildDataIsConsistent_buildDataIsConsistent {
+  dispatch_sync(_db.databaseQueue, ^{
+    EXUpdatesUpdate *update = [EXUpdatesNewUpdate updateWithNewManifest:_manifest response:nil config:_configChannelTest database:_db];
+    
+    NSError *error;
+    [_db addUpdate:update error:nil];
+    XCTAssertNil(error);
+  });
+
+  [EXUpdatesBuildData setBuildDataInDatabase:_db config:_configChannelTest error:nil];
+  
+  NSError *error;
+  [EXUpdatesBuildData ensureBuildDataIsConsistent:_db config:_configChannelTest error:nil];
+  XCTAssertNil(error);
+
+  dispatch_sync(_db.databaseQueue, ^{
+    NSError *error;
+    NSDictionary *staticBuildData = [EXUpdatesBuildData getBuildDataFromDatabase:_db scopeKey:scopeKey error:nil];
+
+    XCTAssertTrue([staticBuildData isEqualToDictionary:[EXUpdatesBuildData getBuildDataFromConfig:_configChannelTest]]);
+    NSArray<EXUpdatesUpdate *> *allUpdates = [_db allUpdatesWithConfig:_configChannelTest error:nil];
+    XCTAssertEqual(allUpdates.count, 1);
+    
+    XCTAssertNil(error);
+  });
+}
+
+- (void)test_ensureBuildDataIsConsistent_buildDataIsInconsistent {
```

Comment:
do you think it's worth separately testing that we get the correct behavior if `releaseChannel` is inconsistent and there's no `channel` defined?